src/outlet/index.ts

```diff
@@ -224,11 +232,44 @@ export class Outlet {
 		const { tokenUrlName, tokenSecretName, tokenIdName, itemStorageKey } = this.tokenConfig
 
 		try {
-			const tokens = readMagicUrl(tokenUrlName, tokenSecretName, tokenIdName, itemStorageKey, this.urlParams)
+			const newToken = readMagicUrl(tokenUrlName, tokenSecretName, tokenIdName, itemStorageKey, this.urlParams)
+			let tokensOutput = readTokens(itemStorageKey)
+
+			let isNewQueryTicket = true
+
+			// loop tokens and remove the tokens having same ids
+			let tokens = []
+			const decodedNewToken = decodeToken(
+				newToken,
+				this.tokenConfig.tokenParser,
+				this.tokenConfig.unsignedTokenDataName,
+				false,
+			)
 
-			await this.whitelistCheck(evtid, 'write')
+			for (const tokenData of tokensOutput.tokens) {
+				if (tokenData.token === newToken.token) {
+					isNewQueryTicket = false
+					break
+				}
+
+				const decodedTokenData = decodeToken(
+					tokenData,
+					this.tokenConfig.tokenParser,
+					this.tokenConfig.unsignedTokenDataName,
+					false,
+				)
+				// hard-coded for temporary solution
+				if (decodedNewToken['ticketIdNumber'] !== decodedTokenData['ticketIdNumber']) {
```

Comment:
As tickets may be stored for multiple conferences (events), we need to look at devconId as well as ticketIdNumber for determining uniqueness.

I suggest you build a unique ID based on both these values, as well as the legacy ticketIdString:
const id = `${decodedToken.devconId}-${decodedToken.ticketIdNumber ?? decodedToken.ticketIdString}`;

Might be best to put it in a separate function to ensure consistency. Once this is done you can use this ID to determine uniqueness rather than just the ticketIdNumber.

Reply:
Yeah, Good point. This is what I missed in the first PR. I thought of this after this PR. I will fix it in next PR